src/main/java/com/synopsys/integration/alert/web/api/role/RoleActions.java

```diff
@@ -35,101 +34,132 @@
-import com.synopsys.integration.alert.common.exception.AlertConfigurationException;
-import com.synopsys.integration.alert.common.exception.AlertDatabaseConstraintException;
-import com.synopsys.integration.alert.common.exception.AlertFieldException;
-import com.synopsys.integration.alert.common.exception.AlertForbiddenOperationException;
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.exception.AlertException;
 import com.synopsys.integration.alert.common.persistence.model.PermissionKey;
 import com.synopsys.integration.alert.common.persistence.model.PermissionMatrixModel;
 import com.synopsys.integration.alert.common.persistence.model.UserRoleModel;
 import com.synopsys.integration.alert.common.rest.model.ValidationResponseModel;
 import com.synopsys.integration.alert.common.security.authorization.AuthorizationManager;
 import com.synopsys.integration.alert.common.util.BitwiseUtil;
+import com.synopsys.integration.alert.component.users.UserManagementDescriptorKey;
+import com.synopsys.integration.alert.web.api.role.util.PermissionModelUtil;
 
 @Component
-public class RoleActions {
+public class RoleActions extends AbstractResourceActions<RolePermissionModel, MultiRolePermissionModel> {
     private static final String FIELD_KEY_ROLE_NAME = "roleName";
     private final RoleAccessor roleAccessor;
     private final AuthorizationManager authorizationManager;
     private final DescriptorMap descriptorMap;
 
     @Autowired
-    public RoleActions(RoleAccessor roleAccessor, AuthorizationManager authorizationManager, DescriptorMap descriptorMap, List<DescriptorKey> descriptorKeys) {
+    public RoleActions(UserManagementDescriptorKey userManagementDescriptorKey, RoleAccessor roleAccessor, AuthorizationManager authorizationManager, DescriptorMap descriptorMap, List<DescriptorKey> descriptorKeys) {
+        super(userManagementDescriptorKey, ConfigContextEnum.GLOBAL, authorizationManager);
         this.roleAccessor = roleAccessor;
         this.authorizationManager = authorizationManager;
         this.descriptorMap = descriptorMap;
     }
 
-    public List<RolePermissionModel> getRoles() {
-        return roleAccessor.getRoles().stream()
-                   .map(this::convertUserRoleModel)
-                   .collect(Collectors.toList());
-    }
-
-    public ValidationResponseModel validateRoleFields(RolePermissionModel rolePermissionModel) {
-        return validateRoleNameFieldRequired(rolePermissionModel.getRoleName())
-                   .map(requiredFieldError -> ValidationResponseModel.fromStatusCollection("There were problems with the role configuration", List.of(requiredFieldError)))
-                   .orElseGet(() -> ValidationResponseModel.success("The role name is valid"));
+    @Override
+    protected ActionResponse<RolePermissionModel> createWithoutChecks(RolePermissionModel resource) {
```

Comment:
I know this change was made a while back but why is every method `...WithoutCheck`?  Is it because the validation is done elsewhere? Why don't we make that more clear without forcing that suffix on every method?